### PR TITLE
Remove grpc-teleterm Make target and Dockerfile-teleterm

### PR DESCRIPTION
**Pull request by ravicious**
grpc-teleterm and Dockerfile-teleterm were originally added because generating JS protobufs on arm64 involved compiling grpc_node_plugin from source. It's a binary that needs a lot of deps and compiling it takes a few minutes even on new MBPs.

In order to not have every dev on arm64 pay this cost just so that the Connect team can generate JS protobufs, we decided to create a separate Dockerfile and use a separate Make target.

It's been a year and Connect became even more reliant on code outside of `lib/teleterm`. We now want to share protos with `lib/prehog` (#19774) and Dockerfile-teleterm started being used to create Linux packages for Connect. Having a separate target doesn't make sense anymore and thanks to what Alan did with libfido2 in the Dockerfile, I found a way to build grpc_node_plugin once and cache the result.

This allows us to remove grpc-teleterm in favor of adding lib/teleterm proto generation to `make grpc`.

---

Since Dockerfile-teleterm was used during tag builds, I made [12.0.0-dev.ravicious.2](https://drone.platform.teleport.sh/gravitational/teleport/19122) that used changes from this PR and it passed successfully. That was before a rebase on master which didn't include some minor changes to JS protos, so I'm running another tag build, [12.0.0-dev.ravicious.3](https://drone.platform.teleport.sh/gravitational/teleport/19128), to verify again that everything's okay.

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,11 @@
 # teleport built on any newer Ubuntu version will not run on Centos 7 because
 # of this.
 
-ARG RUST_VERSION
```
Comment (ravicious) [inline build.assets/Dockerfile:10]: `RUST_VERSION` was defined twice in Dockerfile.

File: build.assets/genproto.sh [hunk 2]
```diff
@@ -26,6 +27,14 @@ main() {
     --path=proto/teleport/lib/multiplexer/
   buf generate --template=lib/prehog/buf.gen.yaml lib/prehog/proto
 
+  # Generate lib/teleterm & JS protos.
+  # TODO(ravicious): Refactor generating JS protos to follow the approach from above, that is have a
+  # separate call to generate Go protos and another for JS protos instead of having
+  # teleterm-specific buf.gen.yaml files.
+  # https://github.com/gravitational/teleport/pull/19774#discussion_r1061524458
+	buf generate --template=lib/prehog/buf-teleterm.gen.yaml lib/prehog/proto
+	buf generate --template=lib/teleterm/buf.gen.yaml lib/teleterm/api/proto
```
Comment (ravicious) [inline build.assets/genproto.sh:36]: Originally I wanted to do this in this PR. But the list of changes is already big enough so I'll handle that in a separate PR (which I'm working on at the moment).

File: build.assets/Dockerfile [hunk 3]
```diff
@@ -282,5 +327,22 @@ RUN cd /usr/local/lib && \
...
+FROM buildbox AS grpc_node_plugin_binary_compiled
+COPY --from=grpc_node_plugin grpc_node_plugin ./grpc_node_plugin
+ONBUILD RUN npm install --global --ignore-scripts grpc-tools@1.11.2 && \
+    mv grpc_node_plugin $(npm root -g)/grpc-tools/bin/. && \
+    # grpc-tools needs protoc but we already install it a couple of steps above.
+    ln -s $(which protoc) $(npm root -g)/grpc-tools/bin/protoc
+
+# A "conditional" FROM like this breaks up the number of steps in the progress bar of the given
+# image, so let's use it at the end. This way the progress bar for buildbox stays mostly untouched.
+FROM grpc_node_plugin_binary_${GRPC_NODE_PLUGIN_BINARY_TYPE} as buildbox
```
Comment (ravicious) [inline build.assets/Dockerfile:345]: This conditional FROM is based on this SO answer: https://stackoverflow.com/a/60820156/742872

**Review by mdwn — APPROVED** [review]

**Comment by ravicious** [comment]
Rebased on master with very minimal changes. During the webapps merge, #20361 was merged as well which also added Node.js to build.assets/Dockerfile so I removed that from my commit.

I'm about to start a new tag build to verify that everything still works.

**Comment by ravicious** [comment]
@fspmarshall Ping.

**Review by fspmarshall — APPROVED** [review]

**Pull request merged**